File: php/kode program/Store.php
<?php
require_once "Electronic.php"; // mengimpor class Electronic sebagai parent

// class Store adalah turunan dari Electronic
class Store extends Electronic {
    // atribut tambahan khusus Store
    private $location;    // lokasi penjual
    private $sellerName;  // nama penjual
    private $contact;     // kontak penjual (nomor hp/wa/dll)

    // constructor untuk inisialisasi objek Store
    public function __construct($id, $name, $price, $stock, $image, $brand, $warranty, $category, $location, $sellerName, $contact) {
        // panggil constructor dari Electronic agar atribut dasar terisi
        parent::__construct($id, $name, $price, $stock, $image, $brand, $warranty, $category);
        // isi atribut tambahan milik Store
        $this->location = $location;
        $this->sellerName = $sellerName;
        $this->contact = $contact;
    }

    // getter
    public function getLocation() { return $this->location; }
    public function getSellerName() { return $this->sellerName; }
    public function getContact() { return $this->contact; }

    // setter
    public function setLocation($location) { $this->location = $location; }
    public function setSellerName($sellerName) { $this->sellerName = $sellerName; }
    public function setContact($contact) { $this->contact = $contact; }
}
?>

File: cpp/kode program/Store.cpp
#include <iostream>
#include "Electronic.cpp"
using namespace std;

// class store merupakan turunan dari electronic
// menambahkan atribut khusus seperti lokasi, nama penjual, dan kontak
class Store : public Electronic{
    private:
        string location;    // menyimpan lokasi toko atau produk
        string sellerName;  // menyimpan nama penjual
        string contact;     // menyimpan kontak penjual

    public:
    // constructor kosong
    Store() {

    }

    // constructor dengan parameter, mewarisi atribut dari electronic
    Store(int id, string name, int price, int stock, string image,
                string brand, string warranty, string category,
                string location, string sellerName, string contact) : Electronic(id, name, price, stock, image, brand, warranty, category){
                this->location = location;
                this->sellerName = sellerName;
                this->contact = contact;
            }

    // setter dan getter untuk location
    void setLocation(string location) {
        this->location = location;
    }
    string getLocation() {
        return this->location;
    }

    // setter dan getter untuk sellerName
    void setSellerName(string sellerName) {
        this->sellerName = sellerName;
    }
    string getSellerName() {
        return this->sellerName;
    }

    // setter dan getter untuk contact
    void setContact(string contact) {
        this->contact = contact;
    }
    string getContact() {
        return this->contact;
    }
};
